### PR TITLE
Add chat teams interop sample

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
Provide a quickstart for how to use calling and chat in the same app to connect to a teams meeting.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: join-chat-to-teams-meeting/Chat_Teams_InteropApp.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//  Chat Teams Interop
+//
+//  Copyright (c) Microsoft Corporation.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+@main
+struct Chat_Teams_InteropApp: App {
```
Comment (palatter) [inline join-chat-to-teams-meeting/Chat_Teams_InteropApp.swift:12]: Is there a reason we didn't use camel case here?

Comment (emlynmac) [reply join-chat-to-teams-meeting/Chat_Teams_InteropApp.swift:12]: This is the default from Xcode. The quickstart is about going form ground zero, with minimal changes to defaults.

File: join-chat-to-teams-meeting/ContentView.swift [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        ) {(call, error) in
+          if let e = error {
+            self.message = "Failed to join call: " + e.localizedDescription
+            print(e.localizedDescription)
+            return
+          }
+
+          self.call = call
+          self.callObserver = CallObserver(self)
+          self.call!.delegate = self.callObserver
```
Comment (jpeng-ms) [inline join-chat-to-teams-meeting/ContentView.swift:181]: Is there a way to not force unwrap?

File: join-chat-to-teams-meeting/ContentView.swift [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+struct MeetingMessage: Identifiable {
+  let id: String
+  let date: Date
+  let content: String
+  let displayName: String
+
+  static func fromTrouter(event: ChatMessageReceivedEvent) -> MeetingMessage {
+    let displayName: String = event.senderDisplayName ?? "Unknown User"
+    let content: String = event.message.replacingOccurrences(
```
Comment (jpeng-ms) [inline join-chat-to-teams-meeting/ContentView.swift:330]: what we are trying to sanitize here?

**Review by palatter — APPROVED** [review]

File: join-chat-to-teams-meeting/Chat Teams Interop.xcworkspace/xcuserdata/emlyn.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JoshuaLai) [inline join-chat-to-teams-meeting/Chat Teams Interop.xcworkspace/xcuserdata/emlyn.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: I don't think we need to check this in

File: join-chat-to-teams-meeting/Chat Teams Interop.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JoshuaLai) [inline join-chat-to-teams-meeting/Chat Teams Interop.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: I don't think we need to check this in.

File: join-chat-to-teams-meeting/Chat Teams Interop.xcodeproj/xcuserdata/emlyn.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JoshuaLai) [inline join-chat-to-teams-meeting/Chat Teams Interop.xcodeproj/xcuserdata/emlyn.xcuserdatad/xcschemes/xcschememanagement.plist:1]: I don't think we need to check this in. Points to your personal user data

**Review by JoshuaLai — COMMENTED** [review]
I think we can just put a git ignore under here: `join-chat-to-teams-meeting/Chat Teams Interop.xcworkspace/xcuserdata/`

File: join-chat-to-teams-meeting/ContentView.swift [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+      .navigationBarTitle("Teams Chat Interop")
+    }
+
+    .onAppear {
+      if let threadIdFromMeetingLink = getThreadId(from: self.meetingLink) {
+        self.chatThreadId = threadIdFromMeetingLink
+      }
+      // Authenticate
+      do {
+        let credentials = try CommunicationTokenCredential(token: token)
```
Comment (JoshuaLai) [inline join-chat-to-teams-meeting/ContentView.swift:108]: Recommendation as a quickstart we should push developers to use `CommunicationTokenCredential(withOptions:)` where they can pass in a refresh token using something like this.
> private func creatTokenRefreshOptions(
        initialToken: String? = nil,
        refreshedToken: String? = nil
    ) -> CommunicationTokenRefreshOptions {
        return CommunicationTokenRefreshOptions(
            initialToken: initialToken,
            refreshProactively: true,
            tokenRefresher: creatTokenRefresher(refreshedToken: refreshedToken)
        )
    }

File: join-chat-to-teams-meeting/ContentView.swift (hunk 3 above)
Comment (emlynmac) [reply join-chat-to-teams-meeting/ContentView.swift:330]: that looks like HTML stripping to me.

File: join-chat-to-teams-meeting/ContentView.swift (hunk 7 above)
Comment (emlynmac) [reply join-chat-to-teams-meeting/ContentView.swift:108]: We can have that as an update later.

**Pull request merged**